internal/domain/amount.go

```diff
@@ -0,0 +1,7 @@
+package domain
+
+type Amount struct {
```

Comment:
This is entirely a matter of taste, but if there is so little difference between entities and domain objects, I tend to not separate them to remove the need for mappings.